### ISSUE TITLE
database wont open!

### ISSUE THREAD

**Issue by Vegan-us** [keepassxreboot/keepassxc#313]
I can open database only every other time.

## Current Behavior
 [Unable to open the database. Wrong key or database file is corrupt.] I keep receiving this message after being able to open it...

## Possible Solution
I wish I knew

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* KeePassXC version/commit used: 2.1.1
* Qt version (e.g. Qt 5.3):5.5.1 
* Compiler (e.g. Clang++3.6.0): 
* Operating System and version:Ubuntu 16.04 LTs

**Comment by phoerious** [comment]
Well, I suppose, you mistyped your password? Or maybe have capslock on? Or wrong keyboard layout? ...?

**Comment by RealOrangeOne** [comment]
Assuming you are typing the password incorrectly, you could try reinstalling keepassxc. Alternatively you could try the _Repair Database_ option under `Database > Repair Database` and see if that does something about it.

**Comment by Vegan-us** [comment]
Hi to all of  you and thank you for your help. For the password I usually copy/paste it, after verifying I was not able to log in getting the same message. Today when I wanted to open my database I got this [Unable to open the database. Wrong key or database file is corrupt.] After a couple of try I finally could log in. The same thing happened with KeepassX by the way. The whole experience make that tool not really reliable to me.

**Comment by phoerious** [comment]
This really makes no sense and I suspect external reasons. The decryption process is deterministic, so given the same input (password and file), it always produces the same output (file can be decrypted or not).
Reasons I could think of are:
- your hard drive is dying (watch for your SMART values, run a thorough disk check, try another hard drive)
- your main memory has errors (run a memcheck, try switching RAM modules)
- some other process is writing to the file (is it on some sort of shared drive?)
- your hard disk firmware is bogus (try saving the file to another drive)
- something else is wrong with your software configuration (try a different PC or operating system, or at least a virtual machine)
- ...

**Comment by droidmonkey** [comment]
I can't imagine this to be a program error. Have yoy tried making a brand new database file to see if you can replicate the issue?

**Comment by phoerious** [comment]
@Vegan-us Was your problem solved? Would be nice to hear what the problem was.

**Comment by Vegan-us** [comment]
@phoerious
Hi, I did not resolved any thing but everything seems to be in order.
I have created a couple of database to test the login along side my original database that I can open now with no trouble.
To resume: Dell Inspiron 15 5k series dual boot. (New)
I have migrate from portable keepass2 windows to Keepassxc ubuntu 16.04LTS via keepassx.
Through all that process I have created new database and imported database which occasionally open or not. At this moment I have two database, #1 now seems to be working, #2 won't open.
Sorry for not being technical this isn't my field. I thought that some users around would have had the same issue. But I believe that I must have a bad understanding of how the .kdbx works...
Thanks again for your help.

**Comment by phoerious** [comment]
Alright, I'll close this then.

**Comment by Vegan-us** [comment]
@phoerious
No, I did not solve any thing but my main database seems to be working.
To resume:
I have migrate from portable keepass2 (Windows) to keepassxc ubuntu via keepassx.
Creating new database and importing others through that process.
At this moment I have two database that I use one is working fine but can't open the other one, nothing vital though.
I must have a bad understanding of how .kdbx files work.
Thanks again for your support.

**Comment by DeanKuai** [comment]
I met this problem too. Here were what i didi (kdbx file is in dropbox folder):
1. open kdbx and add  some new entries.
2. adjust folder structure (create and move).
3. save and quit.
4. try to reopen the kdbx file after about 30m, failed.
5. restore some old versions from dropbox, all can not open.
6. create a new kdbx with the same password, works fine.
7. update keypassxc from 2.1.4 to 2.2.4, still can not open those kdbxs.

Fortunately, I can find those passwords in mac os's keychain.

**Comment by phoerious** [comment]
It's a corruption issue, that means the database becomes unusable. 2.2.4 fixed the reason why this happens, but it cannot make a previously corrupted database readable again.

**Comment by phoerious** [comment]
BTW this more likely related to #1113

**Comment by DeanKuai** [comment]
@phoerious thanks for your reply~

**Comment by chernetskynick** [comment]
Just faced same issue, database corrupted and wont repair. Backup file looks to be corrupted too.

**Comment by chernetskynick** [comment]
thx god google drive keeps backup of each version, but  keepassxc prove to be unreliable, not sure how to trust it any more after this incident.

**Comment by TheZ3ro** [comment]
@chernetskynick what version are you using? 2.2.4 is the latest stable release and it's fixed.
Also, backup files shouldn't be corrupted if they were correctly saved before this issue

**Comment by chernetskynick** [comment]
I just opened backup with version 2.2.2 and seems like it got corrupted as well. Now I can't open it with 2.2.2 neither with 2.2.4.

**Comment by phoerious** [comment]
If it's corrupted, you won't be able to open it with any version. But 2.2.4 should have fixed the cause for corruption. If 2.2.4 still keeps corrupting your DB, it would be great if you could explain in detail how we can reproduce the problem.

**Comment by relin-qq** [comment]
Hi, i have the same problem even on 2.2.4.
I try my best to remember what I did:
- Create db
- Create folder "websites"
- Create several entries for websites (all lower case titles)
- Save
- Create new folder "google"
- Wrongly created as sub-folder of "websites"
- Relocate to root
- Drag & Drop google entries from "websites" to "google"
- Create some more entries in "google"
- Save
- Create new folder "private" with dollar sign as inheritable symbol
- Create new entries "amazon" etc.
- Try to download favicon on each entry (failed on some entries without favicon)
- Save
- Six days later I can't open my database

I hope this helps.
Regards
Relin.

**Comment by renta** [comment]
Faced the same issue on MacOS on both KeePassX and KeePassXC. The database does not open. On ubuntu everything works fine. Double checked and copy-pasted my password from the editor.

**Comment by eshcheglov** [comment]
Same problem here. *All* my databases is corrupted. 

Previously, I use KeepassX on Arch Linux (don't remember the exact version, sorry), and today I installed Fedora 27 to my laptop instead of Arch. Now all my databases won't open with "Unable to open the database.
Wrong key or database file is corrupt.". 

I've copied all kdbx files to the external hard drive, than move it back after reinstallation, and now them don't want to open. What the hell is going on? All other backuped files is OK, only KDBX (not only one, *5* databases) won't open anymore. 

I'm sure that password is correct. I'm almost sure that files was not corrupted (because all other files in backup is good). Version is 2.2.4 from Fedora 27 repo.

**Comment by jlaplante333** [comment]
Everyone I know gets this problem on Mac Os X, and so do I. Database opens perfectly on Windows.

Thanks,

Jon

**Comment by phoerious** [comment]
Can you please try if it happens with a new database well? Also please try our 2.3 beta version.

**Comment by jlaplante333** [comment]
New databases are fine though, works great. It's just any existing Windows databases are always corrupt when I open them. I'm using powershell to back it up constantly because I'm pretty sure it will just become corrupt again.

**Comment by renta** [comment]
Have found out a reason for this strange behavior on MacOS. I have special signs in my password, so clicking on 6 with command key will give you sign ˆ with an underscore. If you will continue to type your password, it will be something like â sign. And if you will not type a letter after compose+6, but type left arrow key - it will be ˆ sign (that's why I had an issue). And only after typing a combination of compose+6 and break after I got a right sign ^. On Ubuntu keyboard works in another way. It produces ^ with shift+6 keys. So, be aware!

**Comment by o2themar** [comment]
I just downloaded KeepassXC on Mac OSX running macOS High Sierra version 10.13.4. I created a new database and setup both password and keyfile login. When I lock the db with no entires and try to login again using either my password or the keyfile I get the error message below. I have tried to create a new database multiple times and in the beginning I keep getting the same error message without adding any entries.

**Error message:**
Unable to open the database.
Wrong key or database file is corrupt.

**Comment by phoerious** [comment]
You always need to use ALL keys you used for creating the database. When you create a database with password and key file, you need to use both to unlock it. Make sure the corresponding checkboxes are ticked on the unlock screen.

**Comment by o2themar** [comment]
@phoerious You are correct. It works! I thought the reason for having the two different authentication methods was to provide two different ways to unlock the database. I was coming from the original Keypass site and saw in their documentation that you can use one or the other to sign in. Maybe they were referring to what you were saying. Thank you for your help.

**Comment by edsu** [comment]
I landed here with the same error. Somehow I had checked the `Key File` box at some point, and once I unchecked it I could open my database again. Whew!

**Comment by notand** [comment]
I had the same error. @phoerious Your hint about the "Key File" box helped. Thanks! To me, this is a GUI design issue. I thought that in the "Key File" field, I had to choose the KeePass file that I want to open (the file that contains all my keys - the ones I want to access). I was not aware that the master key / password may be provided using a "key file".
I am a user who needs to access passwords from keepass files that some other people in my organization provide. I do so only once in a while.

**Comment by droidmonkey** [comment]
@notand looks like the GUI is still misleading you. The key file is not for storing your database password, it is part of the total "key" that is used to decrypt your database. Your database is encrypted using an additive combination of {Password} + {Key File Binary Data} + {Yubikey Challenge Response}. When you check the checkbox, it is telling KeePassXC to use that information to form the master key.

**Comment by notand** [comment]
@phoerious Thanks for the info. In my case, all I need to know is that I must not check the "Key File" box, because my organization does not use this method. My point is that changing the GUI design would help many users out there who just want to access a password without understanding much about key formation. Just after I posted my comment, a colleague of mine told me about having had the same problem.

**Comment by droidmonkey** [comment]
@notand opened #2445 to capture these issues. Feel free to comment there.

**Comment by notand** [comment]
@droidmonkey Great, thanks!

**Comment by alexmaragko** [comment]
I faced the same issues on version 2.4.1, and by not checking the "Key FIle" box, as @notand suggested, I was able to open the database.

**Comment by ramomex1** [comment]
2019 still a bug jajajaa

**Comment by randythomas** [comment]
I encountered this same problem after doing a forced Mac Update.  Is there a solution?  I know I have the right password as I have been using the program for years now.

**Comment by droidmonkey** [comment]
Please read the post. You need to uncheck everything except for the master password box.

**Comment by randythomas** [comment]
OMG!  That worked.  How ridiculous!

Randy Thomas

GPTE IT Operations Manager

Red Hat <https://www.redhat.com/>

Remote: Fort Worth, Tx

althomas@redhat.com
M: 614-315-4516
<https://red.ht/sig>

**Comment by zahardoc** [comment]
Thanks, you saved my life :)

**Comment by samandhare** [comment]
> Please read the post. You need to uncheck everything except for the master password box.



> Please read the post. You need to uncheck everything except for  the master password box.



> Please read the post. You need to uncheck everything except for the master password box.

ohhh great..! 👍  you save my time.

**Comment by CoWinkKeyDinkInc** [comment]
Hi, I haven't opened the database since about June, and I am having problems trying to open it. I don't see anywhere for a checkbox. I have the correct password and key and it still doesn't open, I have a backup database. Was I supposed to upgrade the database from an earlier version? I'm using KeePassXC 2.6.2 on Arch Linux.

**Comment by franziska-haaf** [comment]
> I faced the same issues on version 2.4.1, and by not checking the "Key FIle" box, as @notand suggested, I was able to open the database.

This. It's still a thing end of 2021 even with being such an easy fix. Just bad usability :D

**Comment by michaelk83** [comment]
This has been fixed since 2.5.0, two years ago. Current version is 2.6.6.

**Comment by franziska-haaf** [comment]
> This has been fixed since 2.5.0, two years ago. Current version is 2.6.6.

Ok wow, I have to talk to our Filewave-Admin then 😅 Sorry for the wrong accusation!

**Comment by dimjava** [comment]
Hi there from 2022, had the same problem twice already. First time it was fixed by itself after some time. Waiting what happens this time, but no way I'm using KeepassXC again.

Version 2.7.1, Mac OS 11.4


UPD. Problem fixed after the following: close all previously opened pages with databases, reopen the app, add databases one more time.

**Comment by me00001** [comment]
> UPD. Problem fixed after the following: close all previously opened pages with databases, reopen the app, add databases one more time.

Same here 
Version 2.7.1, Gentoo Linux

**Comment by michaelk83** [comment]
> Same here
> Version 2.7.1, Gentoo Linux

You guys should open a new issue with all the relevant details. It's not likely to be the same bug that was tracked by this issue.
Also, double-check that you're entering the password correctly. If you use a non-standard keyborad layout, there's a known issue in Qt that types some characters wrong when the password is hidden.

**Comment by droidmonkey** [comment]
Please don't open a new issue, we cannot help you when your database is unopenable.
